### DISCUSSION TITLE
How to Add Presets in the UnoCSS Plugin?

### DISCUSSION THREAD

**Discussion by quadratz** [lumeland/lume#584] [Q&A]
I'm encountering an error while attempting to add presets in the UnoCSS plugin. Here's the code snippet I'm using:
```ts
import lume from "lume/mod.ts";
import unocss from "lume/plugins/unocss.ts";
import { presetWind } from 'npm:unocss';

const site = lume();

site.use(unocss({
  options: {
    presets: [
      presetWind({  // <-- Error
        prefix: "un-"
      }),
    ]
  }
}));

export default site;

```

The error message I'm receiving is:

```log
Type 'Preset<Theme>' is not assignable to type 'PresetOrFactory<object> | PresetOrFactory<object>[]'.
```

Could someone please advise on the correct approach to adding UnoCSS presets within the UnoCSS plugin in a Lume project? Any help would be appreciated.

Thanks!

**Comment by oscarotero** [comment]
could you try to import it from `npm:@unocss/preset-wind`?

**Reply by quadratz** [reply]
```ts
import presetWind from "npm:@unocss/preset-wind";
```
I'm getting a different error now:
```log
This expression is not callable.
  Type 'typeof import("file:///C:/Users/dev/unocss/node_modules/.deno/@unocss+preset-wind@0.58.6/node_modules/@unocss/preset-wind/dist/index")' has no call signatures.
```

**Reply by oscarotero** [reply]
I think it should be
```js
import { presetWind } from "npm:@unocss/preset-wind";
```

**Reply by quadratz** [reply]
```ts
import { presetWind } from "npm:@unocss/preset-wind";
```
```log
Type 'Preset<Theme>' is not assignable to type 'PresetOrFactory<object> | PresetOrFactory<object>[]'.
  Type 'Preset<Theme>' is not assignable to type 'Preset<object>'.
    Types of property 'rules' are incompatible.
      Type 'Rule<Theme>[] | undefined' is not assignable to type 'Rule<object>[] | undefined'.
        Type 'Rule<Theme>[]' is not assignable to type 'Rule<object>[]'.
          Type 'Rule<Theme>' is not assignable to type 'Rule<object>'.
            Type '[RegExp, DynamicMatcher<Theme>]' is not assignable to type 'Rule<object>'.
              Type '[RegExp, DynamicMatcher<Theme>]' is not assignable to type '[string, CSSObject | CSSEntries] | [RegExp, DynamicMatcher<object>]'.
                Type '[RegExp, DynamicMatcher<Theme>]' is not assignable to type '[string, CSSObject | CSSEntries]'.
                  Type at position 0 in source is not compatible with type at position 0 in target.
                    Type 'RegExp' is not assignable to type 'string'.
```

**Reply by oscarotero** [reply]
This should work:

```js
import { presetWind } from "npm:@unocss/preset-wind@0.58.5";
```

The problem seems to be caused by different versions of `unocss`. Using the same versions of Lume fixes the issue.

**Reply by quadratz** [reply]
Unfortunately, the same error persists.

**Reply by oscarotero** [reply]
Hmm, it works for me. Are you using the latest Lume version?
If so, do you have a public repo to take a look?

**Reply by quadratz** [reply]
Here's the repo: https://github.com/quadratz/lume-unocss. It seems enabling `nodeModulesDir` in the `deno.json` triggers the error (?)

**Reply by oscarotero** [reply]
I see. I never used `nodeModuleDir` in Deno, so didn't see this bug before.
Out of curiosity, why do you want to use it?

I think this must be a bug from Deno. Looks like something is missing if this option is enabled.

**Reply by quadratz** [reply]
> I think this must be a bug from Deno

I think so.

> I see. I never used nodeModuleDir in Deno, so didn't see this bug before.
Out of curiosity, why do you want to use it?

The [official plugin](https://marketplace.visualstudio.com/items?itemName=antfu.unocss) requires me to install the `node_modules` to make it work. Thus, the temporary solution is to ignore the error by adding `@ts-ignore`.

I've [updated the repo](https://github.com/quadratz/lume-unocss/commit/9e5f64fb1c983a4cb1d620632e4a327d2e730e19) with the code where the plugin works as expected. But I'm not sure if it is the correct way.
